src/main/java/com/projecturanus/betterp2p/client/gui/InfoFilter.kt

```diff
@@ -0,0 +1,78 @@
+package com.projecturanus.betterp2p.client.gui
+
+/**
+ * Extend the default filtering. Holds information about the search filter.
+ *
+ * There are 3 modes of filtering:
+ * - By input/output: Using `@in/@out` filters by input/output respectively.
+ * - By bound/unbound: Use `@b` or `@u` filters by input/output respectively.
+ * - By name: Use the name
+ * If someone comes and says "sort by freq pls" then we can add it at that time
+ */
+class InfoFilter {
+
+    /**
+     * Active filters to use when filtering entries.
+     */
+    val activeFilters: MutableMap<Filter, MutableList<String>?> = mutableMapOf()
```

Comment:
Afaik you can't put nulls as values into maps, even in Kotlin. Nulls and missing values are treated the same, use empty lists or Optionals instead.

Reply:
They should: [see this playground](https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS44LjEwIiwicGxhdGZvcm0iOiJqYXZhIiwiYXJncyI6IiIsIm5vbmVNYXJrZXJzIjp0cnVlLCJ0aGVtZSI6ImlkZWEiLCJjb2RlIjoiZnVuIG1haW4oKSB7XG5cblx0dmFsIG1hcDogTXV0YWJsZU1hcDxTdHJpbmcsIFN0cmluZz8+ID0gbXV0YWJsZU1hcE9mKClcblxuICAgIG1hcFtcIlRlc3RcIl0gPSBudWxsXG4gICAgbWFwW1wiSGVsbG9cIl0gPSBudWxsXG4gICBcdHByaW50bG4obWFwW1wiVGVzdFwiXSlcbiAgICBwcmludGxuKG1hcFtcIkhlbGxvXCJdKVxuICAgIHByaW50bG4obWFwLmtleXMpXG59In0=)

The intended use is that the other filters' values are don't cares.